infra/src/main/kotlin/fi/fta/geoviite/infra/publication/PublicationService.kt

```diff
@@ -707,7 +704,14 @@ class PublicationService @Autowired constructor(
     @Transactional(readOnly = true)
     fun getPublicationDetailsAsTableItems(id: IntId<Publication>): List<PublicationTableItem> {
         logger.serviceCall("getPublicationDetailsAsTableRows", "id" to id)
-        return getPublicationDetails(id).let(::mapToPublicationTableItems)
+        return getPublicationDetails(id).let { publication ->
+            val previousPublication =
+                publicationDao.fetchPublicationTimes().entries.find { it.value < publication.publicationTime }
+            mapToPublicationTableItems(
+                publication,
+                previousPublication?.value ?: publication.publicationTime.minusMillis(1)
```

Comment:
Muutosvertailut tarvivat verrattavan ajanhetken. Yleensä käytetään edellistä julkaisua, mutta jos ollaan vertailemassa ihan ensimmäistä julkaisua, niin sitä ei ole saatavilla. Tällöin defaultataan millisekunnin julkaisua edeltävään aikaan